src/leveldb/db/db_impl.cc

```
@@ -1040,42 +1042,45 @@ Status DBImpl::DoCompactionWork(CompactionState* compact) {
 struct IterState {
-  port::Mutex* mu;
-  Version* version;
-  MemTable* mem;
-  MemTable* imm;
+  port::Mutex* const mu;
+  Version* const version GUARDED_BY(mu);
+  MemTable* const mem GUARDED_BY(mu);
+  MemTable* const imm GUARDED_BY(mu);
+
+  IterState(port::Mutex* mutex, MemTable* mem, MemTable* imm, Version* version)
+      : mu(mutex), version(version), mem(mem), imm(imm) {}
```

cppcheck (2.10-2)

[error] Member variable 'version' is initialized by itself.

(selfInitialization)